pkg/mcs/port_set.go

```diff
@@ -0,0 +1,263 @@
+package mcs
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation"
+	mcsv1a1 "sigs.k8s.io/mcs-api/pkg/apis/v1alpha1"
+)
+
+const (
+	LIGHTHOUSE_PORT_KEY                   = "lighthouse.submariner.io/port"
+	LIGHTHOUSE_PORT_KEY_NAMED_FORMAT      = LIGHTHOUSE_PORT_KEY + ".%s"
+	LIGHTHOUSE_PORT_VALUE_FORMAT_PORT     = "%d"
+	LIGHTHOUSE_PORT_VALUE_FORMAT_PROTOCOL = LIGHTHOUSE_PORT_VALUE_FORMAT_PORT + "-%s"
+	LIGHTHOUSE_PORT_VALUE_FORMAT_FULL     = LIGHTHOUSE_PORT_VALUE_FORMAT_PROTOCOL + "-%s"
+)
+
+// PortSet encapsulates Servie ports used.
+type PortSet struct {
+	ports []mcsv1a1.ServicePort
+}
+
+// NewPortSet creates an empty PortSet.
+func NewPortSet() *PortSet {
+	return &PortSet{}
+}
+
+// NewPortSetFromServicePorts creates a PortSet populated from the ports defined
```

Comment:
I think it would be helpful to provide an example in the docs for how the annotated form looks like

Reply:
agree that example would be useful. Probably easier to extract it from the controllers once they're written. This is the initial version and subject to change... You can also print the test results for now...